### PR TITLE
feat: Add MaxEventSize to limit event size

### LINKED ISSUE #1104: Make MaxEventSize configuration and prevent the too large Event sent out
# 🚀 Feature Request

### Description [**REQUIRED**]
This feature request comes from https://github.com/edgexfoundry/edgex-go/issues/3237
The large event can be prevented from the Device Service side.

### Describe the solution you'd like
Create a MaxEventSize configuration and prevent the event to be sent out if the size is exceed the limit and log a error

### PR DESCRIPTION
**Pull request by jinlinGuan**
fix #1104

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) will update once this PR merged.
  <link to docs PR>

## Testing Instructions

1. build device-simple from this branch and run with core services
2. set maxEventSize to 1 (kb)
3. send GET /api/v2/device/name/Simple-Device01/Image\?ds-pushevent\=yes, and verified the error logs

### CODE REVIEW

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
Please fix the check fail

> internal/common/utils_test.go:38:5: `ctx` is unused (deadcode)
> 
var ctx = context.Background()

File: internal/common/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+)
+
+const (
+	TestDevice = "testDevice"
+	//TestDeviceService  = "testDeviceService"
+	TestProfile        = "testProfile"
+	TestDeviceResource = "testResource"
+	TestDeviceCommand  = "testCommand"
+	//TestValue          = "testValue"
+	//TestUnits          = "testUnits"
```
Comment (cloudxxx8) [inline internal/common/utils_test.go:33]: remove those unused variable comments

Comment (jinlinGuan) [reply internal/common/utils_test.go]: removed

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**